### ISSUE TITLE
Can't yield a function expression without parentheses

### ISSUE THREAD

**Issue by benjamn** [babel/babel#1259]
As demonstrated [here](https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&code=function%20*g()%20%7B%0A%20%20yield%20function()%20%7B%7D%3B%0A%7D%0Ag().next().value()%3B), yielding a function expression triggers an `Unexpected token` error at the `function` keyword:
```js
function *g() {
  yield function() {};
}
g().next().value();
```
Note that enclosing the function expression in parentheses avoids the problem, but the non-parenthesized version works everywhere else (V8/Firefox/Esprima), so I believe it ought to work without parentheses in Acorn.

**Comment by benjamn** [comment]
As you might be able to guess, I'm switching https://github.com/facebook/regenerator over to Acorn, and ran into this problem [here](https://github.com/facebook/regenerator/blob/1bda5d4e4f825b64a773a457d9ce4cf175342668/test/tests.es6.js#L228).

**Comment by sebmck** [comment]
Can you report this upstream to [acorn](https://github.com/marijnh/acorn)?  This was likely a change made in relation to marijnh/acorn#219.

**Comment by benjamn** [comment]
Done: https://github.com/marijnh/acorn/issues/249

**Comment by sebmck** [comment]
Fixed as of 5.1.9.
